.github/workflows/continuous-deployment.yml

```diff
@@ -107,11 +110,9 @@ jobs:
 
   deploy:
     # Only run on main branch
-    if: github.ref == 'refs/heads/main'
     runs-on: ubuntu-latest
-    needs:
-      - build-and-push-image-server
-      - build-and-push-image-client
+    if: inputs.build == true && github.ref == 'refs/heads/main'
```

Comment:
We always want to run the deploy script. this allows us to deploy without building first